File: find-the-middle-index-in-array/find-the-middle-index-in-array.py
class Solution:
    def findMiddleIndex(self, nums: List[int]) -> int:
        i = 0 
        n = len(nums)
        while i<n:
            sum1 , sum2 = 0 ,0 
            for j in range(0,i):
                sum1 += nums[j]
            for k in range(i+1,n):
                sum2 += nums[k]
            if sum1==sum2:
                return i 
            i = i+1
        return -1

File: find-the-middle-index-in-array/find-the-middle-index-in-array.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& nums) {
        int i = 0 ;
        int n = nums.size();
        while(i<n){
            int sum1 = 0 , sum2 = 0 ;
            for(int  j = 0 ; j < i  ; j++){
                sum1 += nums[j];
            }
            for(int k = i+1 ; k < n ; k++){
                sum2 += nums[k];
            }
            if(sum1==sum2){
                return i ;
            }
            i++;
        }
        return -1;
    }
};